docs/core/idioms/file-based.md

```diff
@@ -75,3 +81,20 @@ The flow is ready to run! Every time you need to change the code inside your flo
 ::: warning Flow Structure
 If you change any of the structure of your flow such as task names, rearrange task order, etc. then you will need to reregister that flow.
 :::
+
+
+### File based Docker storage
+
+```python
+flow.storage = Docker(
+    path="my_flow.py",
+    files={"/source/of/my_flow.py": "my_flow.py"},
+    stored_as_file=True
+)
+```
+
+To store flows as files in Docker storage three kwargs needs to be set:
+
+- `path`: pointing to the path that the file is stored in the Docker image
+- `files`: a dictionary of local file source to path destination in image
```

Comment:
Isn't this one only needed if you're using prefect's storage build to store your flow source?

If I use a dockerfile that contains `ADD`/`COPY` lines to add all my flow code, then the `Docker` storage would only need the `path` and the `stored_as_script` kwargs to work correctly?

I could see this being a potentially nice pattern. User writes a dockerfile to add flow source for multiple flows, then uses a storage object that indicates which file in the image to use as the flow source. There's no prefect-side image build step, but the source is still stored in the image.

Reply:
Ah yes this is only needed if you use prefect's default Docker build. I will update this idiom to mention this fact